File: java-source/src/test/java/net/corda/examples/obligation/NodeDriver.java
package net.corda.examples.obligation;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableSet;
import net.corda.core.identity.CordaX500Name;
import net.corda.testing.driver.DriverParameters;
import net.corda.testing.driver.NodeHandle;
import net.corda.testing.driver.NodeParameters;
import net.corda.testing.node.User;

import static net.corda.testing.driver.Driver.driver;

public class NodeDriver {
    public static void main(String[] args) {
        final User user = new User("user1", "test", ImmutableSet.of("ALL"));

        driver(new DriverParameters().setIsDebug(true).setStartNodesInProcess(true).setWaitForAllNodesToFinish(true), dsl -> {
            try {
                NodeHandle nodeA = dsl.startNode(new NodeParameters()
                        .setProvidedName(new CordaX500Name("PartyA", "London", "GB"))
                        .setRpcUsers(ImmutableList.of(user))).get();
                NodeHandle nodeB = dsl.startNode(new NodeParameters()
                        .setProvidedName(new CordaX500Name("PartyB", "New York", "US"))
                        .setRpcUsers(ImmutableList.of(user))).get();
                NodeHandle nodeC = dsl.startNode(new NodeParameters()
                        .setProvidedName(new CordaX500Name("PartyC", "Paris", "FR"))
                        .setRpcUsers(ImmutableList.of(user))).get();

                dsl.startWebserver(nodeA);
                dsl.startWebserver(nodeB);
                dsl.startWebserver(nodeC);
            } catch (Throwable e) {
                System.err.println("Encountered exception in node startup: " + e.getMessage());
                e.printStackTrace();
            }
            return null;
        });
    }
}

File: kotlin-source/src/test/kotlin/net/corda/examples/obligation/NodeDriver.kt
package net.corda.examples.obligation

import net.corda.core.identity.CordaX500Name
import net.corda.core.utilities.getOrThrow
import net.corda.node.services.transactions.SimpleNotaryService
import net.corda.nodeapi.User
import net.corda.nodeapi.internal.ServiceInfo
import net.corda.testing.driver.driver

fun main(args: Array<String>) {
    val user = User("user1", "test", permissions = setOf())

    driver(isDebug = true, startNodesInProcess = true) {
        startNode(
                providedName = CordaX500Name("Controller", "London", "GB"),
                advertisedServices = setOf(ServiceInfo(SimpleNotaryService.type))
        )

        val (nodeA, nodeB, nodeC) = listOf(
                startNode(providedName = CordaX500Name("PartyA", "London", "GB"), rpcUsers = listOf(user)),
                startNode(providedName = CordaX500Name("PartyB", "New York", "US"), rpcUsers = listOf(user)),
                startNode(providedName = CordaX500Name("PartyC", "Paris", "FR"), rpcUsers = listOf(user))
        ).map { it.getOrThrow() }

        startWebserver(nodeA)
        startWebserver(nodeB)
        startWebserver(nodeC)

        waitForAllNodesToFinish()
    }
}